kerlescan/view_helpers.py

```diff
@@ -143,17 +146,22 @@ def ensure_entitled(request, app_name, logger):
     if _is_mgmt_url(request.path) or _is_openapi_url(request.path, app_name):
         return  # allow request
 
-    if auth_key:
-        entitlements = json.loads(base64.b64decode(auth_key)).get("entitlements", {})
-        if entitlement_key in entitlements:
-            if entitlements[entitlement_key].get("is_entitled"):
-                logger.debug("enabled entitlement found on header")
-                return  # allow request
-    else:
-        logger.debug("identity header not sent for request")
+    auth_key = get_key_from_headers(request.headers)
+
+    if not auth_key:
+        logger.debug("entitlement not found for account.")
+        raise HTTPError(HTTPStatus.BAD_REQUEST, message="identity not found on request")
+
+    # check if the request comes from our own drift service
+    if check_request_from_drift_service(request=request):
+        return
+
+    entitlements = json.loads(base64.b64decode(auth_key)).get("entitlements", {})
+    if entitlement_key in entitlements:
+        if entitlements[entitlement_key].get("is_entitled"):
+            logger.debug("enabled entitlement found on header")
+            return  # allow request
 
-    # if we got here, reject the request
-    logger.debug("entitlement not found for account.")
```

Comment:
If we get here, wouldn't we still want to add it to the logger as well as raising the BAD_REQUEST?

Reply:
Good catch! You're right! we definitely need to raise a BAD_REQUEST like `ensure_entitled`